actors/builtin/reward/reward_test.go

```diff
@@ -1 +1,191 @@
 package reward_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/builtin/reward"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestConstructor(t *testing.T) {
+	actor := rewardHarness{reward.Actor{}, t}
+
+	receiver := tutil.NewIDAddr(t, 1000)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+	rt := builder.Build(t)
+	actor.constructAndVerify(rt)
+}
+
+func TestAwardBlockReward(t *testing.T) {
+	actor := rewardHarness{reward.Actor{}, t}
+
+	receiver := tutil.NewIDAddr(t, 1000)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+
+	t.Run("block reward with no payable reward", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		miner := tutil.NewIDAddr(t, 1000)
+		penalty := abi.NewTokenAmount(10)
+		gasreward := abi.NewTokenAmount(10)
+		nominalpwr := abi.NewStoragePower(10)
+		rt.SetBalance(gasreward)
+		actor.blockRewardAndVerify(rt, miner, penalty, gasreward, nominalpwr)
+
+		var st reward.State
+		rt.GetState(&st)
+		rewardMap := adt.AsMultimap(adt.AsStore(rt), st.RewardMap)
+		assert.Equal(t, big.Zero(), st.RewardTotal)
+		var r []reward.Reward
+		var rwd reward.Reward
+		err := rewardMap.ForEach(reward.AddrKey(miner), &rwd, func(i int64) error {
+			r = append(r, rwd)
+			return nil
+		})
+		require.NoError(t, err)
+		// there are 0 rewards in the actors state
+		assert.Equal(t, 0, len(r))
+	})
+
+	t.Run("block reward with reward payable", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		miner := tutil.NewIDAddr(t, 1000)
+		penalty := abi.NewTokenAmount(10)
+		gasreward := abi.NewTokenAmount(10)
```

Comment:
If they're going to cancel out, just use zero for this test (so the reward total is jsut block reward)